sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -250,13 +254,33 @@
      * @throws UnsupportedOperationException If this batch has already added an operation of another type.
      */
     public Response<Void> setBlobAccessTier(String blobUrl, AccessTier accessTier, String leaseId) {
-        return setBlobAccessTierHelper(getUrlPath(blobUrl), accessTier, leaseId);
+        return setBlobAccessTier(new BlobBatchSetBlobAccessTierOptions(blobUrl, accessTier).setLeaseId(leaseId));
     }
 
-    private Response<Void> setBlobAccessTierHelper(String urlPath, AccessTier accessTier, String leaseId) {
+    /**
+     * Adds a set tier operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatch.setBlobAccessTier#BlobBatchSetBlobAccessTierOptions}
+     *
+     * @param options {@link BlobBatchSetBlobAccessTierOptions}
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setBlobAccessTier(BlobBatchSetBlobAccessTierOptions options) {
+        return setBlobAccessTierHelper(options);
+    }
+
+    private Response<Void> setBlobAccessTierHelper(BlobBatchSetBlobAccessTierOptions options) {
```

Comment:
Given this is internal would it be better to splay out the `BlobBatchSetBlobAccessTierOptions` parameters so we don't need to instantiate short lived objects in the other overloads.

Reply:
I'm not too attached to either format but this just keeps things consistent with how we deal with overloads in other places.

Reply:
updated